dev/src/glgeom.cpp

```diff
@@ -134,14 +134,25 @@ void BindAsSSBO(Shader *sh, string_view_nt name, int id) {
 void Mesh::Render(Shader *sh) {
     if (prim == PRIM_POINT) SetPointSprite(pointsize);
     sh->Set();
-    if (numbones && numframes) {
-        int frame1 = ffloor(curanim);
-        int frame2 = frame1 + 1;
-        float frameoffset = curanim - frame1;
-        float3x4 *mat1 = &mats[(frame1 % numframes) * numbones],
-                 *mat2 = &mats[(frame2 % numframes) * numbones];
-        auto outframe = new float3x4[numbones];
-        for(int i = 0; i < numbones; i++) outframe[i] = mix(mat1[i], mat2[i], frameoffset);
+    if (sh->bones_i >= 0 && numbones && numframes) {
+        auto blend_frame = [this](float fr, float3x4 *out) {
+            int frame1 = ffloor(fr);
+            int frame2 = frame1 + 1;
+            float frameoffset = fr - frame1;
+            float3x4 *mat1 = &mats[(frame1 % numframes) * numbones],
+                     *mat2 = &mats[(frame2 % numframes) * numbones];
+            for(int i = 0; i < numbones; i++) out[i] = mix(mat1[i], mat2[i], frameoffset);
+        };
+
+        auto outframe = new float3x4[numbones*2];
```

Comment:
Spaces around operators.

Using unused memory in the base case doesn't feel super clean to me, but I also understand its at virtually no cost since its temp.. so I guess its ok.